M/MuJoCo/build_tarballs.jl

```diff
@@ -0,0 +1,52 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder
+
+name = "MuJoCo"
+version = v"2.3.7"
+
+# Collection of sources required to build libX11
+sources = [
+    GitSource("https://github.com/deepmind/mujoco", "c9246e1f5006379d599e0bcddf159a8616d31441")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/mujoco
+sed -i '/Werror/d' cmake/MujocoOptions.cmake
+CPPFLAGS="-I${prefix}/include"
+CXXFLAGS="-I${prefix}/include"
+apk update
+apk upgrade
+cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release .
+cmake --build .
+cmake --install .
+"""
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [p for p in supported_platforms() if !Sys.isapple(p) || p.tags["arch"] != "aarch64"]
+products = [
+    LibraryProduct("libmujoco", :libmujuco),
+    ExecutableProduct("basic", :mujoco_basic),
+    ExecutableProduct("compile", :mujoco_compile),
+    ExecutableProduct("derivative", :mujoco_derivative),
+    ExecutableProduct("record", :mujoco_record),
+    ExecutableProduct("simulate", :mujoco_simulate),
+    ExecutableProduct("testspeed", :mujoco_testspeed),
+    ExecutableProduct("testxml", :mujoco_testxml),
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    BuildDependency("Xorg_libX11_jll"),
+    BuildDependency("Xorg_xorgproto_jll"),
+    BuildDependency("GLFW_jll"),
+    BuildDependency("Xorg_libXrandr_jll"),
+    BuildDependency("Libglvnd_jll"),
+    BuildDependency("Xorg_libXi_jll"),
+    BuildDependency("Xorg_libXinerama_jll"),
+    BuildDependency("Xorg_libXcursor_jll")
+]
+
+# Build the tarballs.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6", preferred_gcc_version=v"11.1.0")
```

Comment:
That's a really new GCC version, which may limit who can use the resulting JLL (I think?). Is there a specific reason to pick that instead of an older one?

Reply:
The libraries require C++ 17, but it seems like GCC 8+ supports this so perhaps it will work with that? We picked this one as we knew it worked on my own machine.

Reply:
> The libraries require C++ 17

Do you know which features are C++17? The language features can be found in GCC 7, but some library features are GCC 8 or GCC 9.

Reply:
I tried GCC 8 and it wouldn't compile on any platforms, I don't know the specific features that is required, they just say C++ 17 in the docs. Actually, we are having quite a lot of issues cross-compiling for Mac and Windows as their cmake file doesn't seem to handle cross-compilation very well. We know this compiles fine natively on macos and they also have windows binaries too, but we will work on creating the bindings for it, and get back to adding other platform support soon. For now, we have all non-aarch64 builds on Linux.

Reply:
> Actually, we are having quite a lot of issues cross-compiling for Mac and Windows as their cmake file doesn't seem to handle cross-compilation very well.

Just FYI, I mantain the mujoco recipe on conda-forge ( https://github.com/conda-forge/mujoco-feedstock ) and there we are able to cross-compile mujoco on arm64 on Linux and macOS. Unfortunatly we have quite a few local patches at the moment, and perhaps the problem you are seeing could be solved by some of those patches. Indeed we would like to fix upstream as much as possible (see https://github.com/conda-forge/mujoco-feedstock/issues/25 and https://github.com/deepmind/mujoco/pull/937).

Reply:
@traversaro  Thanks! That's very helpful, we'll take a look soon!

Reply:
Here, too, I still wonder what the actual errors are with older GCC... But if works for you as-is for now, let's move ahead anyway (this can be revisited in a future update, if desired)